core/vm/operations_verkle.go

```diff
@@ -111,7 +111,12 @@ func gasSelfdestructEIP4762(evm *EVM, contract *Contract, stack *Stack, mem *Mem
 	beneficiaryAddr := common.Address(stack.peek().Bytes20())
 	contractAddr := contract.Address()
 
-	statelessGas := evm.Accesses.TouchBasicData(contractAddr[:], false, contract.Gas, false)
+	wanted := evm.Accesses.TouchBasicData(contractAddr[:], false, contract.Gas, false)
+	if wanted > contract.Gas {
+		return wanted, nil
+	}
+	statelessGas := wanted
```

Comment:
we could change `TouchBasicData` to return both the wanted and the actually charged